app/models/section.rb

```diff
@@ -86,20 +92,20 @@ def update(attributes)
           slug: slug,
           attachments: attachments,
         )
+      @previous_edition = latest_edition
       @latest_edition = SectionEdition.new(attributes)
-
-      editions.push(@latest_edition)
     end
 
     nil
   end
 
   def published?
-    editions.any?(&:published?)
+    previous_edition && previous_edition.published? ||
+      latest_edition.published?
```

Comment:
I've updated this so that I check `latest_edition.published?` first and I've wrapped the `previous_edition` conditions in parentheses.